infrastructure/github-env-setup.yml

```diff
@@ -0,0 +1,66 @@
+Parameters:
+  GitHubOrg:
+    Type: String
+    Default: aws-samples
+  RepositoryName:
+    Type: String
+    Default: pcluster-manager
+  OIDCProviderArn:
+    Description: Arn for the GitHub OIDC Provider. If left empty a new OIDCProvider will be created.
+    Default: ""
+    Type: String
+
+Conditions:
+  CreateOIDCProvider: !Equals 
+    - !Ref OIDCProviderArn
+    - ""
+
+Resources:
+  PrivateDeployRole:
+    Type: AWS::IAM::Role
+    Description: "Role used to deploy PCM to a private environment (for example, demo)"
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action: sts:AssumeRoleWithWebIdentity
+            Principal:
+              Federated: !If 
+                - CreateOIDCProvider
+                - !Ref GithubOidc
+                - !Ref OIDCProviderArn
+            Condition:
+              StringEquals:
+                token.actions.githubusercontent.com:sub: !Sub repo:${GitHubOrg}/${RepositoryName}:ref:refs/heads/main
+      Policies:
+        - PolicyName: UpdateLambdaFromPrivateECR
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - lambda:UpdateFunctionCode
+                  - ecr:DescribeRepositories
+                Resource:
+                  - !Sub arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:PClusterManagerFunction
```

Comment:
For these and other ARNs we should use the [pseudo parameter reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html#cfn-pseudo-param-partition) for the _partition_ as it might [change](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) depending on the region.